dali/operators/decoder/nvjpeg/decoupled_api/nvjpeg_decoder_decoupled_api_test.cc

```diff
@@ -180,4 +181,63 @@ TYPED_TEST(nvjpegDecodeDecoupledAPITest, TestSingleTiffDecode4T) {
   this->TiffTestDecode(4);
 }
 
+class HwDecoderUtilizationTest : public ::testing::Test {
+ public:
+  void SetUp() final {
+    dali::string list_root(testing::dali_extra_path() + "/db/single/jpeg");
+
+    pipeline_.AddOperator(
+            OpSpec("FileReader")
+                    .AddArg("device", "cpu")
+                    .AddArg("file_root", list_root)
+                    .AddOutput("compressed_images", "cpu")
+                    .AddOutput("labels", "cpu"));
+    auto decoder_spec =
+            OpSpec("ImageDecoder")
+                    .AddArg("device", "mixed")
+                    .AddArg("output_type", DALI_RGB)
+                    .AddArg("hw_decoder_load", .7f)
+                    .AddInput("compressed_images", "cpu")
+                    .AddOutput("images", "gpu");
+    pipeline_.AddOperator(decoder_spec, decoder_name_);
+
+    pipeline_.Build(outputs_);
+
+    auto node = pipeline_.GetOperatorNode(decoder_name_);
+    if (!node->op->GetDiagnostic<bool>("using_hw_decoder")) {
+      try {
+        if (nvml::HasHwDecoder()) {
+          FAIL() << "HW Decoder exists in the system and failed to open";
+        }
+      } catch (const std::runtime_error &e) {
```

Comment:
I just hope that FAIL() doesn't work through an exception that's compatible with runtime_error.

Reply:
On the other hand, you could probably check if the functions were loaded -> if the driver has insufficient version they stay as nullptrs, if there was error when loading you get an error thrown. So you can either:
* check if the functions are available (not nullptr), if not skip this test without resorting to exectpions
* extend DALIError_t to something indicating NO_FUNCTION_TO_CALL and work with this in `HasHwDecoder`. Still no random exceptions being thrown around.
extend dali

Reply:
Done